### DISCUSSION TITLE
running as a service

### DISCUSSION THREAD

**Discussion by lfoppiano** [microsoft/BitNet#131] [Ideas]
Hi, first of all, than you for this tool, it's a very useful and interesting approach for running models on low resources. 

I was wondering whether you have any plans to add a way to run it as a service, where the whole model is not loaded every time a new prompt is provided. Something like [llama-server](https://github.com/lfoppiano/llama.cpp?tab=readme-ov-file)? 

I did try to run a model quantized for BitNET with `llama-server` but it seems they are not compatible. Do you have any comment or suggestions? 

Thank you in advance 
Luca

**Comment by celsowm** [comment]
Any news about it?

**Comment by caramdache** [comment]
Look here: https://github.com/microsoft/BitNet/issues/206#issuecomment-2820070694
